pkg/apis/core/validation/validation.go

```diff
@@ -1533,6 +1533,34 @@ func validateStorageOSPersistentVolumeSource(storageos *core.StorageOSPersistent
 	return allErrs
 }
 
+// ValidatePVSecretReference check whether provided SecretReference object is valid in terms of secret name and namespace.
+// This also take a flag to validate the secret name against DNSSubDomainformat (<253char) and DNS1123Label (<63char) format
+// as its required by some PV sources today, ex: CSI. This flag can help on racheting validation of secretRef field.
+func ValidatePVSecretReference(secretRef *core.SecretReference, allowDNSSubDomainFormat bool, fldPath *field.Path) field.ErrorList {
```

Comment:
Could you add a unit test?

Reply:
sure, we do have good amount of unit tests for CSI volume source now (Recently added) , however added one speficially for this func too.. thanks @roycaihw for the review, ptal..